### ISSUE TITLE
azurerm_arc_kubernetes_cluster_extension conflict error when both release_train and version are set

### ISSUE THREAD

**Issue by agreaves-ms** [hashicorp/terraform-provider-azurerm#23690]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.2

### AzureRM Provider Version

3.76.0

### Affected Resource(s)/Data Source(s)

azurerm_arc_kubernetes_cluster_extension

### Terraform Configuration Files

```hcl
resource "azurerm_arc_kubernetes_cluster_extension" "apim" {
  cluster_id       = local.cluster_id
  extension_type   = "Microsoft.ApiManagement.Gateway"
  name             = "api-arc-${var.name}"
  release_train    = "preview"
  version          = "1.6.4"
  target_namespace = "apim"

  identity {
    type = "SystemAssigned"
  }

  configuration_protected_settings = {
    "gateway.auth.token"      = var.gateway_auth_key
    "monitoring.ingestionKey" = ""
  }

  configuration_settings = {
    "service.type"                                                        = "LoadBalancer"
    "service.http.port"                                                   = "8080"
    "service.http.nodePort"                                               = ""
    "service.https.port"                                                  = "8081"
    "gateway.id"                                                          = local.gateway_id
    "gateway.configuration.uri"                                           = "api-${var.name}-002.configuration.azure-api.net"
    "gateway.configuration.backup.enabled"                                = true
    "gateway.configuration.backup.persistentVolumeClaim.storageClassName" = ""
    "telemetry.metrics.opentelemetry.enabled"                             = false
    "telemetry.metrics.opentelemetry.collector.uri"                       = ""
    "replicaCount"                                                        = "1"
    "monitoring.workspaceId"                                              = ""
    "monitoring.customResourceId"                                         = local.apim_service_id
    "highAvailability.enabled"                                            = true
    "dapr.enabled"                                                        = false
    "dapr.app.id"                                                         = ""
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: Conflicting configuration arguments
│ 
│   with azurerm_arc_kubernetes_cluster_extension.apim,
│   on main.tf line 18, in resource "azurerm_arc_kubernetes_cluster_extension" "apim":
│   18:   release_train    = "preview"
│ 
│ "release_train": conflicts with version
╵
╷
│ Error: Conflicting configuration arguments
│ 
│   with azurerm_arc_kubernetes_cluster_extension.apim,
│   on main.tf line 19, in resource "azurerm_arc_kubernetes_cluster_extension" "apim":
│   19:   version          = "1.6.4"
│ 
│ "version": conflicts with release_train
╵
```

### Expected Behaviour

It should create the extension in the following way:

- Allow `Release Train` to be something other than `stable`, in this case, `preview`.
- Pin a `version`, in the case, `1.6.4`
- Set `Auto Upgrade Minor Version` to `False`.

As an example, the following *az cli* command works in the previously described manor:

```shell
az k8s-extension create -g rg-awg-aio-tst --cluster-name arc-awg-aio-tst --cluster-type connectedClusters --extension-type Microsoft.ApiManagement.Gateway --name api-arc-awg-aio-tst --release-train preview --version 1.6.4 --auto-upgrade false --config-file arc-apim-settings.json --config-protected-file arc-apim-protected-settings.json
```

Which produces the following extension in the portal:

<img width="1673" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/111466195/8fcc711c-1f14-4c45-914e-df842e32f520">

### Actual Behaviour

The `azurerm` terraform provider produces a Conflict error when `version` and `release_train` are set at the same time.

**Comment by ms-zhenhua** [comment]
Hi @agreaves-ms, thank you for reaching out.
According to this [MS-DOC](https://learn.microsoft.com/en-us/rest/api/kubernetesconfiguration/extensions/create?tabs=HTTP#request-body), `releaseTrain` works only when `autoUpgradeMinorVersion` is `true` and `version` works only when `autoUpgradeMinorVersion` is `false`. So, `releaseTrain` and `version` cannot be set simultaneously. Have you tried to only set `version=1.6.4` w/o setting `release_train`?

**Comment by stephybun** [comment]
Closed by #23692
